### PR TITLE
Refactor script a bit

### PR DESCRIPTION
**Pull request by fbartels**
Hi,

I have spent some time refactoring codimd a while ago. I have not rebased this on the current version, but since I do not have time at the moment to end my refactoring I wanted to open a pr now.

I do have a problem with the code at the moment, since logins (for my ldap backend) seem to fail.

### CODE REVIEW

**Comment by pirate** [comment]
I don't have LDAP set up, but maybe @SISheogorath can test this branch to see if it works against his LDAP-enabled CodiMD server.  I'm eager to merge this since there are a lot of improvements, I just don't want to break LDAP for existing users if it's still untested/buggy with these changes.

**Comment by SISheogorath** [comment]
Seems like a simple copy & paste mistake:

```patch
diff --git a/bin/codimd b/bin/codimd
index 53b757e..31c927e 100755
--- a/bin/codimd
+++ b/bin/codimd
@@ -94,7 +94,7 @@ function authenticate_email() {
 function authenticate_ldap() {
     curl -c $CODIMD_COOKIES_FILE \
         -XPOST \
-        --data-urlencode "email=$1" --data-urlencode "password=$2" \
+        --data-urlencode "username=$1" --data-urlencode "password=$2" \
         $CODIMD_SERVER/auth/ldap &>/dev/null
 }
 

```

**Comment by SISheogorath** [comment]
Also just as an idea, maybe add something like `CODIMD_DEBUG` which triggers `set -x` ;)

**Comment by fbartels** [comment]
> Seems like a simple copy & paste mistake

Hmm.. Looking at the diff this must have been broken before as well. I'll give it a quick try and push a fix if this is indeed the culprit.

**Comment by SISheogorath** [comment]
:+1: Great work! Really look forward to have this merged. I'll add some CI stack later this weekend and during the next week.

File: bin/codimd [hunk 1]
```diff
@@ -179,12 +189,15 @@ function get_history() {
...
+    noteid=${1:-""}
+    if [[ -z "$noteid" ]]; then
         echo "ERROR: You must specify a note id to delete."
         echo ""
         echo "Usage: codimd delete <note_id>"
         exit 1
-    elif [[ "$(is_authenticated)" ]]; then
+    fi
+
+    if [[ "$(is_authenticated)" ]]; then
```
Comment (SISheogorath) [inline bin/codimd]: Shouldn't `if is_authenticated; then` work as well?

Comment (fbartels) [reply bin/codimd]: yes, that would work as well.

**Comment by SISheogorath** [comment]
I'll leave this to @pirate now :) LGTM

**Comment by fbartels** [comment]
@pirate I'd say it is good for now.

**Pull request merged**